var/spack/repos/builtin/packages/moab/package.py

```diff
@@ -55,61 +78,78 @@ class Moab(Package):
     # depends_on('cgns', when='+cgns')
     # depends_on('vtk', when='+vtk')
 
-    depends_on('mpi')
-    depends_on('hdf5+mpi')
+    depends_on('mpi', when='+mpi')
+    depends_on('hdf5', when='+hdf5')
+    depends_on('hdf5+mpi', when='+hdf5+mpi')
     depends_on('netcdf', when='+netcdf')
-    depends_on('netcdf+mpi', when='+netcdf')
-    depends_on('metis')
-    depends_on('parmetis')
-    depends_on('zoltan')
-    depends_on('zoltan~fortran', when='~fortran')
+    depends_on('parallel-netcdf', when='+pnetcdf')
+    depends_on('cgm', when='+cgm')
+    depends_on('metis', when='+metis')
+    depends_on('parmetis', when='+parmetis')
+    # FIXME it seems that zoltan needs to be built without fortran
+    depends_on('zoltan~fortran', when='+zoltan')
 
-    def install(self, spec, prefix):
+    def configure_args(self):
+        spec = self.spec
 
         options = [
-            '--prefix=%s' % prefix,
             '--enable-optimize',
-            '--disable-tools',
-            '--disable-mbconvert',
-            '--disable-hexmodops',
             '--disable-vtkMOABReader',
-            '--disable-mbsize',
-            '--disable-mbskin',
             '--disable-mbtagprop',
             '--disable-mbmem',
             '--disable-spheredecomp',
             '--disable-mbsurfplot',
-            '--disable-mbpart',
-            '--disable-dagmc',
             '--disable-gsets',
-            '--disable-mbmerge',
-            '--disable-mbdepth',
-            '--disable-mbcoupler',
             '--disable-mcnpmit',
             '--disable-refiner',
             '--disable-h5mtools',
             '--disable-mbcslam',
-            '--disable-mbquality',
-            '--disable-ahf',
-            '--disable-mbumr',
-            '--disable-imesh',
             '--with-pic',
-            '--with-mpi=%s' % spec['mpi'].prefix,
-            '--with-hdf5=%s' % spec['hdf5'].prefix,
-            '--with-parmetis=%s' % spec['parmetis'].prefix,
-            '--with-zoltan=%s' % spec['zoltan'].prefix,
-            '--without-vtk',
-            'CXX=%s' % spec['mpi'].mpicxx,
-            'CC=%s' % spec['mpi'].mpicc,
-            'FC=%s' % spec['mpi'].mpifc]
+            '--without-vtk'
+            ]
+        if '+mpi' in spec:
+            options.extend([
+                '--with-mpi=%s' % spec['mpi'].prefix,
+                'CXX=%s' % spec['mpi'].mpicxx,
+                'CC=%s' % spec['mpi'].mpicc,
+                'FC=%s' % spec['mpi'].mpifc
+                ])
```

Comment:
Flake8 doesn't like this. The closing brackets should be indented the same as the start of the `options` line. You went one level too far.